src/web/pool.rs

```diff
@@ -1,10 +1,12 @@
 use crate::db::create_pool;
-use iron::{typemap, BeforeMiddleware, IronResult, Request};
+use iron::{status::Status, typemap, BeforeMiddleware, IronError, IronResult, Request};
 use postgres::Connection;
+use std::marker::PhantomData;
 
 #[cfg(test)]
 use std::sync::{Arc, Mutex, MutexGuard};
 
+#[derive(Debug, Clone)]
 pub(crate) enum Pool {
```

Comment:
Won't this `Clone` still create more pools? I thought the idea was to limit the number of concurrent connections by using a single pool that kept track of them itself.

Reply:
`Pool<T>` is just an [`Arc`](https://docs.rs/r2d2/0.8.8/src/r2d2/lib.rs.html#317), so it clones that